sources/web/datalab/auth.ts

```diff
@@ -194,6 +194,20 @@ function setOauth2Client(request: http.ServerRequest): void {
   }
 }
 
+function redirect(response: http.ServerResponse, referer: string) {
+  if (referer == 'popup') {
+    // If auth is triggered from a pop-up window, close the window.
```

Comment:
Assuming 'popup' is the state set by Colab? If so add that to comment so easier to understand and maintain?